### DISCUSSION TITLE
Types are generated twice in v3, making them all duplicated.

### DISCUSSION THREAD

**Discussion by PhilipAngelinNE** [dotansimha/graphql-code-generator#8996] [Q&A] [answered]
Title. In previous v2 versions there were no issues with duplicate types, but now when trying v3 it happens. This is the config:

```ts
import type { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: 'http://localhost:11500/graphql',
  documents: 'api/queries/**/*.graphql',
  ignoreNoDocuments: true,
  generates: {
    'api/gql/': {
      preset: 'client',
      plugins: ['typescript'],
      config: {
        maybeValue: 'T | null | undefined',
      },
    },
  },
};

export default config;
```

Anyone else who's run into this issue? In v3? I found one other thread but that was using the yml config, which worked, so doesn't help here sadly.

**Comment by n1ru4l** [comment, marked as answer]
Hey, @PhilipAngelinNE You should remove the `'typescript'` plugin from the `plugins` array. The client preset already automatically applies this plugin.

```diff
import type { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: 'http://localhost:11500/graphql',
  documents: 'api/queries/**/*.graphql',
  ignoreNoDocuments: true,
  generates: {
    'api/gql/': {
      preset: 'client',
-     plugins: ['typescript'],
+     plugins: [],
      config: {
        maybeValue: 'T | null | undefined',
      },
    },
  },
};

export default config;
```

**Reply by PhilipAngelinNE** [reply]
Thanks for quick reply, that did fix it. But now the watch script only generates that one file, the other 3 are gone, including the one with the helper function. Is there a plugin I need to use for those?

**Edit:** Nevermind after another restart it works as normal again. Thanks for the help once again!

**Reply by n1ru4l** [reply]
There is https://github.com/dotansimha/graphql-code-generator/issues/8963 for the latter issue 😁
